packages/core/chrome/core-chrome-browser-internal/src/chrome_service.test.tsx

```diff
@@ -203,37 +203,10 @@ describe('start', () => {
 
     it('renders the default project side navigation', async () => {
       const { chrome } = await start();
-
       chrome.setChromeStyle('project');
-
       const component = mount(chrome.getHeaderComponent());
-
-      const projectHeader = findTestSubject(component, 'kibanaProjectHeader');
+      const projectHeader = findTestSubject(component, 'kibanaProjectHeaderInvisible');
```

Comment:
@sebelga @Dosant @vadimkibana @clintandrewhall 

I'm stuck on something here. Now that the project header can be hidden by a passed-in observable, I'm not able to come up with how to get a non-hidden header component from the `getHeaderComponent` call. 

I reached out to Tina H about it and she gave me the advice that _the core service is not intended for rendering, that's what the rendering service is for._  That made me think we may want to relocate the logic for choosing the actual header component to the rendering service. If we do that though, we would test the rendering service against a mocked core service. 

Any ideas about what needs to be done to access a "visible" header component in this test?

Reply:
> the core service is not intended for rendering, that's what the rendering service is for. That made me think we may want to relocate the logic for choosing the actual header component to the rendering service instead of having chrome.getHeaderComponent do that.

If I am not wrong we inherited this from the core team 😊 Yes it would be a better separation of concern but I am all about _small_ increments vs big refactors.

First regarding the implementation, instead of passing `isVisible$={isVisible$}` I prefer the parent to have the switch logic.

```ts
const isVisible = somLogic === 'visible';

return isVisible ? <ComponentFoo /> : <ComponentBar />
```

For the test of the visibility you need to set an application as active. Looking at the code (not tested) this is how I would do it

```ts
    it('renders the default project side navigation', async () => {
      const { chrome } = await start({
        startDeps: defaultStartDeps([{ id: 'foo', title: 'Foo' } as App], 'foo'), // pass one application + the active application id
      });
     ...
```

Update the `defaultStartDeps` to take an active app id

```ts
function defaultStartDeps(availableApps?: App[], currentAppId?: string) {
  const deps = {
    application: applicationServiceMock.createInternalStartContract(currentAppId), // and pass it here
    docLinks: docLinksServiceMock.createStartContract(),
    ...
```

Update the`createInternalStartContract`

```ts
const createInternalStartContractMock = (
  currentAppId?: string
): jest.Mocked<InternalApplicationStart> => {
  const currentAppId$ = new BehaviorSubject<string | undefined>(currentAppId); // Change "Subject" to "BehaviorSubject"

  return {
    applications$: new BehaviorSubject<Map<string, PublicAppInfo>>(new Map()),
```